File: replicate/bugSolved.java
import java.text.DecimalFormat;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class bugSolved {

    // improved version of the bug.java hehe
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<Integer> temperature = new ArrayList<>();

        System.out.println("Temperature Analyzer");
        System.out.println("Enter the temperature for each day:");

        for (int i = 1; i <= 7; i++) {
            System.out.print("Day " + i + ": ");
            String input = scanner.nextLine().trim();

            int temp = Integer.parseInt(input);
            temperature.add(temp);

        }

        int sum = 0;
        
        for (int temp : temperature) {
            sum += temp;
        }

        double average = (double) sum / temperature.size();
        DecimalFormat formattedAverage = new DecimalFormat("#.00");

        System.out.println("Weekly average: " + formattedAverage.format(average) + " °C");

        int max = Collections.max(temperature);
        System.out.println("Hottest Temperature: " + max + " °C");

        scanner.close();
    }
}

File: replicate/bugSolved.py


#improvised version saa bug.py replaced day with temperature para di confusing

def main():
    print("Temperature Analyzer")
    print("Enter the temperature for each day:")

    temperature = []

 
    for i in range(1, 8):
        temp = input(f"Day {i}: ").strip()
        temperature.append(int(temp))

    sum_temp = 0
    max_temp = -999


    for t in temperature:

        sum_temp += t 
        if t > max_temp:
            max_temp = t  

    average = sum_temp / len(temperature)

    print(f"Weekly average: {average:.2f} °C")
    print(f"Hottest Temperature: {max_temp} °C")

if __name__ == "__main__":
    main()
